File: Week 1/id_455/LeetCode_283_455.py
#
# @lc app=leetcode.cn id=283 lang=python3
#
# [283] 移动零
#

# @lc code=start
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        for num in nums[:]:
            if num == 0:
                nums.append(0)
                nums.remove(0)
# @lc code=end

File: Week 1/id_455/LeetCode_283_455.c
/*
 * @lc app=leetcode.cn id=283 lang=c
 *
 * [283] 移动零
 */

// @lc code=start


void moveZeroes(int* nums, int numsSize){
    int j=0;
    for(int i = 0; i < numsSize; ++i){
        if(nums[i] != 0){
            nums[j] = nums[i];
            if(i != j){
                nums[i] = 0;
            }
            ++j;
        }
    }
}


// @lc code=end

